### PR TITLE
Delete deployment with proper delete propagation policy

### PR DESCRIPTION
**Pull request by life1347**
1. Migrate DeleteOptions `OrphanDependents` to `PropagationPolicy`. 
2. Remove legacy code (workaround) for old k8s version.

---
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/630)

### CODE REVIEW

File: executor/cleanup.go [hunk 1]
```diff
@@ -65,26 +69,14 @@ func cleanup(client *kubernetes.Clientset, namespace string, instanceId string)
 	if err != nil {
 		return err
 	}
-	// See K8s #33845 and related bugs: deleting a deployment
-	// through the API doesn't cause the associated ReplicaSet to
-	// be deleted.  (Fixed recently, but we may be running a
-	// version before the fix.)
-	err = cleanupReplicaSets(client, namespace, instanceId)
-	if err != nil {
```
Comment (vishal-biyani) [inline executor/cleanup.go]: I am sorry but why did we remove the error check here?

**Comment by vishal-biyani** [comment]
Looks good except for 2 minor comments @life1347 
I don't think we have a good way to test this automatically though ATM. Can we have a placeholder bug for future?

File: executor/cleanup.go (hunk 1 above)
Comment (life1347) [reply executor/cleanup.go]: The bug described in the comment was fixed in k8s 1.6. Since fission required 1.7+ , remove the legacy code here.

**Comment by life1347** [comment]
https://github.com/fission/fission/issues/658

File: executor/cleanup.go (hunk 1 above)
Comment (life1347) [reply executor/cleanup.go]: I guess you mean L84, fixed.

**Pull request merged**